src/blockmap.jl

```diff
@@ -297,75 +297,96 @@ LinearAlgebra.transpose(A::BlockMap) = TransposeMap(A)
 LinearAlgebra.adjoint(A::BlockMap)  = AdjointMap(A)
 
 ############
-# multiplication with vectors
+# multiplication helper functions
 ############
 
-function A_mul_B!(y::AbstractVector, A::BlockMap, x::AbstractVector)
-    require_one_based_indexing(y, x)
-    m, n = size(A)
-    @boundscheck (m == length(y) && n == length(x)) || throw(DimensionMismatch("A_mul_B!"))
+@inline function _blockmul!(y, A::BlockMap, x, α, β)
```

Comment:
This used to be the `A_mul_B!` code, which can be easily generalized to work with `α`, `β`, and matrices instead of vectors, so I factored it out. The generic version of indexing is then `selectdim(y, 1, ...)`.